src/test.jl

```diff
@@ -0,0 +1,612 @@
+
+Find an appropriate tolerance for given points or tangent vectors, or their types.
+"""
+find_eps(x...) = find_eps(Base.promote_type(map(number_eltype, x)...))
+find_eps(x::Type{TN}) where {TN<:Number} = eps(real(TN))
+find_eps(x) = find_eps(number_eltype(x))
+
+
+"""
+    test_manifold(
+        M::Manifold,
+        pts::AbstractVector;
+        args,
+    )
+
+Test general properties of manifold `M`, given at least three different points
+that lie on it (contained in `pts`).
+
+# Arguments
+- `default_inverse_retraction_method = ManifoldsBase.LogarithmicInverseRetraction()` - default method for inverse retractions ([`log`](@ref))
+- `default_retraction_method = ManifoldsBase.ExponentialRetraction()` - default method for retractions ([`exp`](@ref))
+- `exp_log_atol_multiplier = 0`, change absolute tolerance of exp/log tests (0 use default, i.e. deactivate atol and use rtol)
+- `exp_log_rtol_multiplier = 1`, change the relative tolerance of exp/log tests (1 use default). This is deactivated if the `exp_log_atol_multiplier` is nonzero.
+- `retraction_atol_multiplier = 0`, change absolute tolerance of (inverse) retraction tests (0 use default, i.e. deactivate atol and use rtol)
+- `retraction_rtol_multiplier = 1`, change the relative tolerance of (inverse) retraction tests (1 use default). This is deactivated if the `exp_log_atol_multiplier` is nonzero.
+- `inverse_retraction_methods = []`: inverse retraction methods that will be tested.
+- `mid_point12 = shortest_geodesic(M, pts[1], pts[2], 0.5)`, if not `nothing`, then check that `mid_point(M, pts[1], pts[2])` is approximately equal to `mid_point12`.
+- `point_distributions = []` : point distributions to test
+- `projection_tvector_atol_multiplier = 0` : chage absolute tolerance in testing projections (0 use default, i.e. deactivate atol and use rtol)
+-  tvector_distributions = []` : tangent vector distributions to test
+- `basis_types = ()` : basis types that will be tested
+- `rand_tvector_atol_multiplier = 0` : chage absolute tolerance in testing random vectors (0 use default, i.e. deactivate atol and use rtol)
+  random tangent vectors are tangent vectors
+- `retraction_methods = []`: retraction methods that will be tested.
+- `test_forward_diff = true`: if true, automatic differentiation using
+  ForwardDiff is tested.
+- `test_reverse_diff = true`: if true, automatic differentiation using
+  ReverseDiff is tested.
+- `test_musical_isomorphisms = false` : test musical isomorphisms
+- `test_mutating_rand = false` : test the mutating random function for points on manifolds
+- `test_project_tangent = false` : test projections on tangent spaces
+- `test_representation_size = true` : test repersentation size of points/tvectprs
+- `test_tangent_vector_broadcasting = true` : test boradcasting operators on TangentSpace
+- `test_vector_transport = false` : test vector transport
+- `test_vector_spaces = true` : test Vector bundle of this manifold
+
+"""
+function test_manifold(
```

Comment:
We should use this opportunity to check that all keywords are well documented, useful and complete. For example I recently added the `is_tangent_atol_multiplier` when working on a manifold with a student.

Reply:
I've added brief descriptions of all keyword arguments and sorted them alphabetically.